wireshark/Dockerfile

```diff
@@ -1,75 +1,146 @@
-   sed -i "s/#include \"config.h\"/#include \"config.h\"\n#include \"qsc.h\"/g" epan/dissectors/packet-tls-utils.c && \
-   sed -i "s/oid_add_from_string(\"sha224\", \"2.16.840.1.101.3.4.2.4\");/oid_add_from_string(\"sha224\", \"2.16.840.1.101.3.4.2.4\");\nQSC_SIGS/g" epan/dissectors/packet-pkcs1.c && \
-   sed -i "s/    { 260\, \"ffdhe8192\" }\, \/\* RFC 7919 \*\//    { 260\, \"ffdhe8192\" }\, \/\* RFC 7919 \*\/\nQSC_KEMS/g" epan/dissectors/packet-tls-utils.c && \
-   sed -i "s/ { 0x080b\, \"rsa_pss_pss_sha512\" }\,/ { 0x080b\, \"rsa_pss_pss_sha512\" }\,\nQSC_SIG_CPS/g" epan/dissectors/packet-tls-utils.c
-
-# Build wireshark
-RUN mkdir -p build && cd build && cmake -GNinja -DCMAKE_INSTALL_PREFIX=/opt/wireshark .. && ninja && ninja install
-
-FROM ubuntu
-ENV DEBIAN_FRONTEND noninteractive
-
-RUN apt update && apt upgrade -y && apt install -y qtbase5-dev qtchooser qt5-qmake qtbase5-dev-tools libc-ares2 libqt5multimedia5 pcaputils libssh-dev
-
-# Only retain the ${INSTALLDIR} contents in the final image
-COPY --from=intermediate /opt/wireshark /opt/wireshark
-
-
-CMD /opt/wireshark/bin/wireshark
+ARG INSTALLDIR
+
+LABEL version="2"
+
+# Install essential build dependencies
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    build-essential libtool automake autoconf cmake ninja-build \
+    openssl libssl-dev git wget ca-certificates  \
+    python3 python3-pip python3-venv && \
+    apt-get clean && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /opt
+# Set up isolated directories
+# src for source files, build for compiling, and install for final binaries
+RUN mkdir -p src/liboqs src/openssl src/oqs-provider src/wireshark \
+    build/liboqs build/openssl build/oqs-provider build/wireshark \
+    ${INSTALLDIR}/lib ${INSTALLDIR}/bin ${INSTALLDIR}/ssl
+
+# Download sources
+WORKDIR /opt/src
+RUN git clone --depth 1 https://github.com/open-quantum-safe/liboqs.git liboqs && \
+    git clone --depth 1 https://github.com/openssl/openssl.git openssl && \
+    git clone --depth 1 https://github.com/open-quantum-safe/oqs-provider.git oqs-provider && \
+    wget -O wireshark.tar.xz https://www.wireshark.org/download/src/all-versions/wireshark-${WIRESHARK_VERSION}.tar.xz && \
+    tar -xf wireshark.tar.xz --strip-components=1 -C wireshark && \
+    rm wireshark.tar.xz
+
+# Build and install liboqs
+WORKDIR /opt/build/liboqs
+RUN cmake -G Ninja /opt/src/liboqs \
+    -D CMAKE_INSTALL_PREFIX=${INSTALLDIR}/liboqs \
+    -D BUILD_SHARED_LIBS=ON \
+    -D OQS_USE_OPENSSL=OFF \
+    -D CMAKE_INSTALL_RPATH="${INSTALLDIR}/liboqs/lib" && \
+    ninja -j$(nproc) && ninja install
+
+# Build OpenSSL integrated with liboqs
+WORKDIR /opt/build/openssl
+RUN LDFLAGS="-Wl,-rpath,${INSTALLDIR}/liboqs/lib" \
+    /opt/src/openssl/config \
+        --prefix=${INSTALLDIR}/openssl \
+        --openssldir=${INSTALLDIR}/ssl \
+        shared && \
+    make -j$(nproc) && \
+    make install_sw install_ssldirs
```

Comment:
Without testing it myself, I'm pretty sure this Dockerfile will fail on arm64 systems (like macOS). This can be addressed by symlinking lib and lib64 for openssl as seen in the [openssl3 demo Dockerfile](https://github.com/open-quantum-safe/oqs-demos/blob/b04c59e4a438a2ae0c704b5af115abacde3652da/openssl3/Dockerfile#L54-L55)

You should be able to test this by running the Docker build and run steps with `--platform linux/amd64,linux/arm64` as detailed in the [docker docs](https://docs.docker.com/build/building/multi-platform/), though I believe it requires installing specific docker support to do so (usually included by default in Win/macOS installs)